crates/services/consensus_module/poa/src/service.rs

```diff
@@ -140,29 +148,65 @@ pub struct Task<T, B, I> {
     consensus_params: ConsensusParameters,
 }
 
-impl<T, B, I> Task<T, B, I>
+impl<T, B, I> MainTask<T, B, I>
 where
     T: TransactionPool,
 {
-    pub fn new(
+    pub fn new<P: P2pPort>(
         last_block: &BlockHeader,
         config: Config,
         txpool: T,
         block_producer: B,
         block_importer: I,
+        p2p_port: P,
+        block_rx: broadcast::Receiver<Arc<ImportResult>>,
     ) -> Self {
         let tx_status_update_stream = txpool.transaction_status_events();
         let (request_sender, request_receiver) = mpsc::channel(100);
         let last_timestamp = last_block.time();
         let duration =
             Duration::from_secs(Tai64::now().0.saturating_sub(last_timestamp.0));
         let last_block_created = Instant::now() - duration;
+
+        // todo get from config
+        let min_conneected_reserved_peers = 0;
+        let time_until_sync = Duration::ZERO;
+
+        let initial_sync_state = {
```

Comment:
You can set it `NotSynced` by default, and later it will be updated by the `SyncTask`

Reply:
While I changed a bit this code, wasn't the idea that if we start with 0 desired connected peers and time until sync is 0 - that we already consider to be Synced.